### PR TITLE
Enable global propagator for AWS instrumentation

### LINKED ISSUE #2598: Enable global propagator for AWS instrumentation
This is a continuation of #2573. After that PR got meged, we now need to refactor the AWS instrumentation to use the AWS Lambda propagator from the global propagator.

### PR DESCRIPTION
**Pull request by ocelotl**
Fixes #2598

### CODE REVIEW

File: instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py [hunk 1]
```diff
@@ -167,7 +168,7 @@ def _determine_parent_context(
     """
     parent_context = None
 
-    if not disable_aws_context_propagation:
+    if "_X_AMZN_TRACE_ID" in os.environ.get(OTEL_PROPAGATORS, ""):
```
Comment (tylerbenson) [inline instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py:171]: No, this should go away completely as the env var check will be done by the propagator instead.

File: instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py [hunk 2]
```diff
@@ -167,13 +167,12 @@ def _determine_parent_context(
     """
     parent_context = None
 
-    if not disable_aws_context_propagation:
-        xray_env_var = os.environ.get(_X_AMZN_TRACE_ID)
+    xray_env_var = os.environ.get(_X_AMZN_TRACE_ID)
```
Comment (tylerbenson) [inline instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py:170]:
```suggestion
```

File: instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py [hunk 3]
```diff
@@ -167,13 +167,12 @@ def _determine_parent_context(
...
 
-        if xray_env_var:
-            parent_context = AwsXRayPropagator().extract(
-                {TRACE_HEADER_KEY: xray_env_var}
-            )
+    if xray_env_var:
+        parent_context = AwsXRayPropagator().extract(
+            {TRACE_HEADER_KEY: xray_env_var}
+        )
```
Comment (tylerbenson) [inline instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py:175]:
```suggestion
```

File: instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py [hunk 4]
```diff
@@ -167,13 +167,12 @@ def _determine_parent_context(
...
 
     if (
         parent_context
```
Comment (tylerbenson) [inline instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py:178]: You might need to do something here to ensure the event context extractor isn't invoked if there's already an active span context.

**Review by tylerbenson — COMMENTED** [review]
I think they just need to be deleted.

File: instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py (hunk 4 above)
Comment (ocelotl) [reply instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py]: Right ,fixed

**Comment by xrmx** [comment]
Why this change? There's not a single text line in the issue, commit or PR :sweat_smile:

**Review by tylerbenson — APPROVED** [review]
CC @rapphil in case you want to take a look too.

**Comment by ocelotl** [comment]
> Why this change? There's not a single text line in the issue, commit or PR 😅

Sorry, I updated the issue with a description :v:

File: instrumentation/opentelemetry-instrumentation-aws-lambda/tests/test_aws_lambda_instrumentation_manual.py [hunk 5]
```diff
@@ -56,6 +56,7 @@ def __init__(self, aws_request_id, invoked_function_arn):
 )
 
 MOCK_XRAY_TRACE_ID = 0x5FB7331105E8BB83207FA31D4D9CDB4C
+
```
Comment (xrmx) [inline instrumentation/opentelemetry-instrumentation-aws-lambda/tests/test_aws_lambda_instrumentation_manual.py:59]:
```suggestion
```

File: instrumentation/opentelemetry-instrumentation-aws-lambda/tests/test_aws_lambda_instrumentation_manual.py [hunk 6]
```diff
@@ -81,6 +82,7 @@ def mock_execute_lambda(event=None):
     """Mocks the AWS Lambda execution.
 
     NOTE: We don't use `moto`'s `mock_lambda` because we are not instrumenting
+
```
Comment (xrmx) [inline instrumentation/opentelemetry-instrumentation-aws-lambda/tests/test_aws_lambda_instrumentation_manual.py:85]:
```suggestion
```

File: propagator/opentelemetry-propagator-aws-xray/src/opentelemetry/propagators/aws/aws_xray_propagator.py [hunk 7]
```diff
@@ -328,9 +328,9 @@ def fields(self):
         return {TRACE_HEADER_KEY}
 
 
-class AwsXrayLambdaPropagator(AwsXRayPropagator):
+class AwsXRayLambdaPropagator(AwsXRayPropagator):
```
Comment (xrmx) [inline propagator/opentelemetry-propagator-aws-xray/src/opentelemetry/propagators/aws/aws_xray_propagator.py:331]: We should probably add a test that asserts that the entry points are loadable

Comment (ocelotl) [reply propagator/opentelemetry-propagator-aws-xray/src/opentelemetry/propagators/aws/aws_xray_propagator.py:331]: Right, fixed

File: CHANGELOG.md [hunk 8]
```diff
@@ -11,6 +11,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - `opentelemetry-instrumentation-flask` Add `http.route` and `http.target` to metric attributes
   ([#2621](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2621))
+- `opentelemetry-instrumentation-aws-lambda` Enable global propagator for AWS instrumentation
+  ([#2708](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2708))
```
Comment (emdneto) [inline CHANGELOG.md:15]:
```suggestion
  ([#2599](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2599))
```

**Review by emdneto — APPROVED** [review]

**Review by rapphil — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-aws-lambda/pyproject.toml [hunk 9]
```diff
@@ -33,6 +33,9 @@ dependencies = [
 [project.optional-dependencies]
 instruments = []
 
+[project.entry-points.opentelemetry_instrumentor]
+aws_lambda= "opentelemetry.instrumentation.aws_lambda:AwsLambdaInstrumentor"
```
Comment (xrmx) [inline instrumentation/opentelemetry-instrumentation-aws-lambda/pyproject.toml:37]:
```suggestion
aws_lambda = "opentelemetry.instrumentation.aws_lambda:AwsLambdaInstrumentor"
```

File: propagator/opentelemetry-propagator-aws-xray/src/opentelemetry/propagators/aws/aws_xray_propagator.py (hunk 7 above)
Comment (xrmx) [reply propagator/opentelemetry-propagator-aws-xray/src/opentelemetry/propagators/aws/aws_xray_propagator.py:331]: I meant for the propagator entry point

Comment (ocelotl) [reply propagator/opentelemetry-propagator-aws-xray/src/opentelemetry/propagators/aws/aws_xray_propagator.py:331]: Added a test

File: instrumentation/opentelemetry-instrumentation-aws-lambda/tests/test_aws_lambda_instrumentation_manual.py [hunk 10]
```diff
@@ -513,3 +518,15 @@ def test_no_op_tracer_provider(self):
         spans = self.memory_exporter.get_finished_spans()
         assert spans is not None
         self.assertEqual(len(spans), 0)
+
+    def test_load_entry_point(self):
+        self.assertIs(
+            next(
+                iter(
+                    entry_points(
+                        group="opentelemetry_instrumentor", name="aws_lambda"
```
Comment (emdneto) [inline instrumentation/opentelemetry-instrumentation-aws-lambda/tests/test_aws_lambda_instrumentation_manual.py:527]:
```suggestion
                        group="opentelemetry_instrumentor", name="aws-lambda"
```

**Pull request merged**